Add metadata for Quartz 2.5.0

Our [Quartz smoke test](https://github.com/spring-projects/spring-aot-smoke-tests/tree/main/framework/quartz) fails because of missing Quartz metadata.

We've worked around that by using `buildArgs.addAll('-H:+UnlockExperimentalVMOptions', '-H:Preserve=package=org.quartz.*')`.

We're using `org.quartz-scheduler:quartz:2.5.0`.

```
Caused by: org.graalvm.nativeimage.MissingReflectionRegistrationError: Cannot reflectively invoke method 'public void org.quartz.impl.jdbcjobstore.JobStoreSupport.setInstanceName(java.lang.String)'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

  {
    "type": "org.quartz.impl.jdbcjobstore.JobStoreSupport",
    "methods": [
      {
        "name": "setInstanceName",
        "parameterTypes": [
          "java.lang.String"
        ]
      }
    ]
  }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.reportInvokedExecutable(MissingReflectionRegistrationUtils.java:110)
	at java.base@25/java.lang.reflect.Method.acquireMethodAccessor(Method.java:144)
	at java.base@25/java.lang.reflect.Method.invoke(Method.java:562)
	at org.quartz.impl.SchedulerDetailsSetter.set(SchedulerDetailsSetter.java:73)
	at org.quartz.impl.SchedulerDetailsSetter.setDetails(SchedulerDetailsSetter.java:46)
	at org.quartz.impl.StdSchedulerFactory.instantiate(StdSchedulerFactory.java:886)
	at org.quartz.impl.StdSchedulerFactory.getScheduler(StdSchedulerFactory.java:1570)
	at org.springframework.scheduling.quartz.SchedulerFactoryBean.createScheduler(SchedulerFactoryBean.java:669)
	at org.springframework.scheduling.quartz.SchedulerFactoryBean.prepareScheduler(SchedulerFactoryBean.java:605)
	at org.springframework.scheduling.quartz.SchedulerFactoryBean.afterPropertiesSet(SchedulerFactoryBean.java:492)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1865)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1814)
	... 18 more
```

--- a/metadata/org.quartz-scheduler/quartz/2.3.2/reflect-config.json
+++ b/metadata/org.quartz-scheduler/quartz/2.3.2/reflect-config.json
@@ -104,5 +104,19 @@
       "typeReachable": "org.quartz.impl.jdbcjobstore.JobStoreSupport"
     },
     "allPublicMethods": true
+  },
+  {
+    "name": "org.quartz.impl.jdbcjobstore.JobStoreTX",
+    "condition": {
+      "typeReachable": "org.quartz.impl.StdSchedulerFactory"
+    },
+    "allDeclaredConstructors": true
+  },
+  {
+    "name": "org.quartz.impl.jdbcjobstore.JobStoreSupport",
+    "condition": {
+      "typeReachable": "org.quartz.impl.StdSchedulerFactory"
+    },
+    "allPublicMethods": true
   }
 ]

--- a/tests/src/org.quartz-scheduler/quartz/2.3.2/src/test/java/quartz/QuartzTest.java
+++ b/tests/src/org.quartz-scheduler/quartz/2.3.2/src/test/java/quartz/QuartzTest.java
@@ -18,6 +18,7 @@ import org.quartz.SimpleScheduleBuilder;
 import org.quartz.Trigger;
 import org.quartz.TriggerBuilder;
 import org.quartz.impl.StdSchedulerFactory;
+import org.quartz.impl.jdbcjobstore.JobStoreTX;
 import org.quartz.simpl.SimpleThreadPool;
 
 import static org.assertj.core.api.Assertions.assertThat;
@@ -74,6 +75,20 @@ public class QuartzTest {
         scheduler.scheduleJob(jobDetail, trigger);
 
         Thread.sleep(100);
+
+        scheduler.shutdown();
+    }
+
+    @Test
+    void jobStoreTX() throws SchedulerException {
+        Properties properties = new Properties();
+        properties.put(StdSchedulerFactory.PROP_JOB_STORE_CLASS, JobStoreTX.class.getName());
+        properties.put("org.quartz.jobStore.dataSource", "ds");
+        properties.put(StdSchedulerFactory.PROP_THREAD_POOL_PREFIX + ".threadCount", Integer.toString(2));
+        StdSchedulerFactory factory = new StdSchedulerFactory(properties);
+        Scheduler scheduler = factory.getScheduler();
+
+        scheduler.shutdown();
     }
 
 }

